x-pack/plugins/ml/common/types/anomaly_detection_jobs/job.ts

```diff
@@ -8,7 +8,7 @@
 import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 
 export type JobId = string;
-export type BucketSpan = string;
+export type BucketSpan = estypes.Duration;
```

Comment:
The bucket span type is a string.
`Duration` also allows `0` and `-1` which are not compatible with the backend.

Reply:
The same goes for `frequency` and `query_delay` in `MlDatafeed` and `time_span` in `MlChunkingConfig`.
I think it's safe to assume all changes where `Duration` is being used for [time units](https://www.elastic.co/guide/en/elasticsearch/reference/master/api-conventions.html#time-units) in ML are incorrect.

I'm assuming this change has been added to bring the client's types inline with es types? where `Duration` is the same as the java `TimeValue`? 
In the case of `BucketSpan`, the elasticsearch endpoint will reject anything that is not a string and will not accept string versions of `"0"` or `"-1"`, it will only accept a "time unit" e.g. `"2d"`

Reply:
> In the case of BucketSpan, the elasticsearch endpoint will reject anything that is not a string and will not accept string versions of "0" or "-1", it will only accept a "time unit" e.g. "2d"

Not according to the client's TS types definition that are coming from ES directly.

Reply:
But as described in the `To codeowners` section, if you think that's a mistake, please feel free to push to the PR to revert my changes and use `@ts-expect-error` instead in the places surfacing violations because of the mismatch with the client's declared types.

Reply:
cc @elastic/clients-team, apparently the changes in types to define these fields as `Duration` are wrong?

Reply:
Which APIs are you referring to @jgowdyelastic?

@pgayvallet I'm wondering if we need to change `BucketSpan` type in Kibana's code to be `estypes.Duration` here, if the `string` type was working before this upgrade then any APIs using the new `estypes.Duration` type should continue accepting strings?

Reply:
>  if the string type was working before this upgrade then any client APIs using the estypes.Duration type should continue accepting strings?

@sethmlarson These conversions are performed in both direction in our codebase

In one way ( string => Duration ) it's fine
In the other way ( Duration => string ) it's an error.

Reply:
>Which APIs are you referring to @jgowdyelastic? It's hard to tell which Elasticsearch API is impacted since the change is made to a type that's used elsewhere.

Any api which deals with anomaly detection job configs, e.g, to put a config: https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-put-job.html

I think I understand how this change has come about after talking to @droberts195 about how es uses the `TimeValue` type for these fields but then has additional restrictions on the values contained inside.

From the point of view of a consumer of the javascript client and the supplied types, using `Duration` for these fields is incorrect. It lets me use values in a payload which will instantly be rejected by elasticsearch when they are sent.
As I mentioned above, I believe this is the same for all of the changes in this client up date where `Duration` is replacing `string` for ML objects. e.g. `frequency`, `query_delay`, `time_span`, there may be more.

I think this change needs to happen in the client. And so to unblock this PR I can accept the `Duration` changes and all the `@ts-expect-error` comments that are needed too make it work. but it would be good to address this in the future and get the correct time unit type in the client.